manual_flagging/initial_flagging.py

```diff
@@ -121,153 +140,375 @@
-df_flagged_ptax_merged = pd.concat(
-    [df_res_flagged_updated_ptax, df_condo_flagged_updated_ptax]
-).reset_index(drop=True)
+if inputs["manual_update"] == True:
+    # Group the existing data by 'ID' and find the maximum 'version' for each sale
+    existing_max_version = (
+        df_flag_table.groupby("meta_sale_document_num")["version"]
+        .max()
+        .reset_index()
+        .rename(columns={"version": "existing_version"})
+    )
 
-# Finish flagging and subset to write to flag table
-df_to_write, run_id, timestamp = flg.finish_flags(
-    df=df_flagged_ptax_merged,
-    start_date=inputs["time_frame"]["start"],
-    manual_update=False,
-)
+for df_name, df in dfs_to_finalize.items():
+    df_copy = df["df"].copy()
 
-# Write to sale.flag table
-flg.write_to_table(
-    df=df_to_write,
-    table_name="flag",
-    s3_warehouse_bucket_path=os.getenv("AWS_S3_WAREHOUSE_BUCKET"),
-    run_id=run_id,
-)
+    df_copy, run_id, timestamp = flg.finish_flags(
+        df=df_copy, start_date=inputs["time_frame"]["start"]
+    )
 
-# Write to sale.parameter table
-df_parameters = flg.get_parameter_df(
-    df_to_write=df_to_write,
-    df_ingest=df_ingest,
-    iso_forest_cols=inputs["iso_forest"],
-    res_stat_groups=inputs["stat_groups"],
-    condo_stat_groups=condo_stat_groups,
-    dev_bounds=inputs["dev_bounds"],
-    ptax_sd=inputs["ptax_sd"],
-    rolling_window=inputs["rolling_window_months"],
-    date_floor=inputs["time_frame"]["start"],
-    short_term_thresh=flg_model.SHORT_TERM_OWNER_THRESHOLD,
-    min_group_thresh=inputs["min_groups_threshold"],
-    run_id=run_id,
-)
+    if inputs["manual_update"] == True:
+        # Merge, compute new version, and drop unnecessary columns
+        df_copy = (
+            df_copy.merge(existing_max_version, on="meta_sale_document_num", how="left")
+            .assign(
+                version=lambda x: x["existing_version"]
+                .apply(lambda y: y + 1 if pd.notnull(y) else 1)
+                .astype(int)
+            )
+            .drop(columns=["existing_version"])
+        )
+
+    dfs_to_write[df_name]["df"] = df_copy
+    dfs_to_write[df_name]["run_id"] = run_id
+    dfs_to_write[df_name]["timestamp"] = timestamp
+
+for df_name, df in dfs_to_write.items():
+    print(f"{df_name}\n")
+    print(dfs_to_feature_creation[df_name]["df"].columns)
+
+    df_parameter = flg.get_parameter_df(
+        df_to_write=dfs_to_write[df_name]["df"],
+        df_ingest=dfs_to_feature_creation[df_name]["df"],
+        iso_forest_cols=dfs_to_write[df_name]["iso_forest_cols"],
+        stat_groups=dfs_to_write[df_name]["columns"],
+        dev_bounds=inputs["dev_bounds"],
+        ptax_sd=inputs["ptax_sd"],
+        rolling_window=inputs["rolling_window_months"],
+        date_floor=inputs["time_frame"]["start"],
+        short_term_thresh=flg_model.SHORT_TERM_OWNER_THRESHOLD,
+        min_group_thresh=inputs["min_groups_threshold"],
+        run_id=dfs_to_write[df_name]["run_id"],
+    )
```

Comment:
I think the group means could very simply just be concatenated together 

I like that idea, but I *think* there are downsides to both. One large motivation for these metadata tables is the ability to have a granular look at how a sale came to be flagged, whether it be historically or as the most recent iteration. In the current model that I'm thinking about (one `run_id` per flagged group of data, one row per parameter/metadata table), I think it is slightly easier to do this. 

As a theoretical example: Lets say there is a sale I need to look at, and I need to figure out why it was flagged as an outlier. First I look at the `run_id`, then I reference `run_id` the `sale.parameter` table to check the grouping columns for explainability 

**comment not finished**

Reply:
I like the way you're thinking about the tradeoffs here! But I'm not totally convinced that the multi-row solution incurs substantially greater precision, and I'm not yet sure how important that precision actually is.

My understanding is that in the multi-row scenario, we would still need to figure out the tri and market type for a sale before we knew what stat groupings applied to it, since in neither solution is there a direct reference between a sale and the parameters that applied to it (we only have `run_id`, which would not uniquely identify a `parameter` row under the multi-row solution, and would not uniquely identify an entry in the nested `stat_groups` data structure under the single-row solution). But maybe I'm misunderstanding your proposal for a unique key; happy to discuss this more.

My other reason for leaning towards a single-row solution is that I don't yet have a ton of confidence that it's mission-critical for these tables to be as precise as possible. As long as we can quickly diagnose the parameters that generated an individual sale, we should be fine; both of these solutions fit that bill in my book, in which case I lean toward the simpler solution since I expect it to be easier to maintain going forward (less data and a simpler data model).

Perhaps we could discuss on a call today or tomorrow?